wandb/sdk/wandb_verify.py

```diff
@@ -0,0 +1,495 @@
+        print_results(failed_test_strings, False)
+        return
+    if set(os.listdir(multi_art_dir)) != set(
+        [
+            "verify_a.txt",
+            "verify_2.txt",
+            "verify_1.txt",
+            "verify_3.txt",
+            "verify_int_test.txt",
+        ]
+    ):
+        failed_test_strings.append(
+            "Artifact directory is missing files. Contact W&B for support."
+        )
+
+    computed = wandb.Artifact("computed", type="dataset")
+    computed.add_dir(multi_art_dir)
+    verify_digest(download_artifact, computed, failed_test_strings)
+
+    computed_manifest = computed.manifest.to_manifest_json()["contents"]
+    downloaded_manifest = download_artifact._load_manifest().to_manifest_json()[
+        "contents"
+    ]
+    verify_manifest(downloaded_manifest, computed_manifest, failed_test_strings)
+
+    print_results(failed_test_strings, False)
+
+
+def check_graphql_put(api: Api, host: str) -> Optional[str]:
+    # check graphql endpoint using an upload
+    print("Checking signed URL upload".ljust(72, "."), end="")
+    failed_test_strings = []
+    gql_fp = "gql_test_file.txt"
+    f = open(gql_fp, "w")
+    f.write("test2")
+    f.close()
+    with wandb.init(project=PROJECT_NAME, config={"test": "put to graphql"}) as run:
+        run_id = run.id
+        entity = run.entity
+        saved, status_code, url = try_manual_save(api, gql_fp, run.id, run.entity)
+        if not saved:
+            print_results(
+                "Server failed to accept a graphql put request with response {}. Check bucket permissions.".format(
+                    status_code
+                ),
+                False,
+            )
+
+            # next test will also fail if this one failed. So terminate this test here.
+            return None
+
+    # wait for upload to finish before download
+    time.sleep(2)
+
+    public_api = wandb.Api()
+    try:
+        prev_run = public_api.run("{}/{}/{}".format(entity, PROJECT_NAME, run_id))
+    except Exception:
+        failed_test_strings.append(
+            "Unable to access previous run through public API. Contact W&B for support."
+        )
+        print_results(failed_test_strings, False)
+        return None
+    try:
+        read_file = prev_run.file(gql_fp).download(replace=True)
+    except Exception:
+        failed_test_strings.append(
+            "Unable to read file successfully saved through a put request. Check SQS configurations, topic configs and SNS configs."
+        )
+        print_results(failed_test_strings, False)
+        return None
+    contents = read_file.read()
+    try:
+        assert contents == "test2"
+    except AssertionError:
+        failed_test_strings.append(
+            "Read file contents do not match saved file contents. Contact W&B for support."
+        )
+
+    print_results(failed_test_strings, False)
+    return url
+
+
+def check_large_file(api: Api, host: str) -> None:
```

Comment:
supernit: technically this isn't a file upload. A better name might be "check_large_post"

Reply:
Agreed, meant to change that.